pagopa/io-app
# feat: [IOCOM-499] SEND selected payment update after selection

## Short description
This PR adds the automatic update of a SEND payment, after having selected the same.
|Select a payment|Navigate and go back|Selected payment is updating|
|-|-|-|
|![Simulator Screenshot - iPhone 14 - 2023-10-30 at 17 14 05](https://github.com/pagopa/io-app/assets/5150343/5eace3e8-7f12-4252-91d4-59a8aff355ae)|![Simulator Screenshot - iPhone 14 - 2023-10-30 at 17 14 17](https://github.com/pagopa/io-app/assets/5150343/901fab9c-754d-4117-b331-ac009b8e95e5)|![Simulator Screenshot - iPhone 14 - 2023-10-30 at 17 14 20](https://github.com/pagopa/io-app/assets/5150343/4d37025c-5849-4de3-8820-bd37d698783f)|

## List of changes proposed in this pull request
- SEND payments' reducer has a new property, `selectedPayment`, an optional `string`. This is the payment Id of the last selected payment that caused a transition to the wallet flow (in order to pay it);
- Two new actions, one to set and one to clear such property and a selector to retrieve it;
- When selecting a payment, the `setSelectedPayment` action is dispatch (to set the payment Id in the respective reducer);
- Main SEND's screen uses an `onFocus` hook to check the `selectedPayment` (without causing a re-rendering) and, if one is set, it clears it and it dispatches an action to update the payment.

## How to test
Using the _io-dev-api-server_ with a proper SEND configuration check that, after having selected a payment and having went back to the SEND message's screen, such payment goes back to the loading state and it is later updated.
